modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +85,31 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = ::Proxy::Dns::Plugin.settings.dns_zones
+      found_match = false
+      wight = 0
+      matched_zone = fqdn.sub(/[^.]+./,'')
+
+      if dns_zones.is_a?(Array)
+        dns_zones.each do |zone|
+          zone.sub!(/^\./, '')
```

Comment:
You can do this check without a regex:

zone_labels = zone.split(".").reverse
fqdn_labels = fqdn.split(".")
found_match = zone_labels.all? { |zone_label| zone_label == fqdn_labels.pop }